docs/internal-documentation/distrib-testing/Ubuntu-build-environment.md

```diff
@@ -0,0 +1,16 @@
+# Ubuntu Build Environment
+
+These are instructions for creating a build environment for Zonemaster
+components based on Perl. This is not meant as instructions for installing
+Zonemaster itself. 
+
+This instruction is for creating it on Ubuntu. See other files for other OSs.
+
+> Should this be Ubuntu 20.04? Do we need to cover more?
+
+> Is git installed by default?
+>
+> Here we need gettext. Anything else?
```

Comment:
This maybe depends on what component we are building.

#### Building Zonemaster-Engine

To build Zonemaster-Engine on a fresh Ubuntu 16.04 server, it seems that you need to install the same dependencies as in the [Zonemaster-Engine Installation document](https://github.com/matsduf/zonemaster-engine/blob/develop/docs/Installation.md#installation-on-debian) :

```
sudo apt install autoconf automake build-essential cpanminus libclone-perl libdevel-checklib-perl libemail-valid-perl libfile-sharedir-perl libfile-slurp-perl libidn11-dev libintl-perl libio-socket-inet6-perl libjson-pp-perl liblist-moreutils-perl liblocale-msgfmt-perl libmodule-find-perl libmodule-install-xsutil-perl libmoose-perl libmoosex-singleton-perl libnet-ip-perl libpod-coverage-perl libreadonly-xs-perl libssl-dev libtest-differences-perl libtest-exception-perl libtest-fatal-perl libtest-pod-perl libtext-csv-perl libtool m4
```

Maybe we don't need all of them but I didn't take the time to dig deeper into it. 

You also need to install `Module::Install`, as required in [Makefile.PL](https://github.com/matsduf/zonemaster-engine/blob/develop/Makefile.PL)

```
sudo cpanm Module::Install
```

And of course `gettext`

```
sudo apt install gettext
```

#### Building the other components

I'll have a look into that as I try to build them on Ubuntu.

Reply:
That would be great. I'll make some updates of the document, and you could check. All dependencies for Engine are not needed for the build server. On the other hand, is "build-essential" really needed for installation of Engine?

Reply:
Yes you need `build-essential` because it gives you the `make` command which is required to install the `Module::Install` and its dependencies.

```
$ sudo cpanm Module::Install
--> Working on Module::Install
Fetching http://www.cpan.org/authors/id/E/ET/ETHER/Module-Install-1.19.tar.gz ... OK
==> Found dependencies: YAML::Tiny
--> Working on YAML::Tiny
Fetching http://www.cpan.org/authors/id/E/ET/ETHER/YAML-Tiny-1.73.tar.gz ... OK
Configuring YAML-Tiny-1.73 ... OK
! Can't configure the distribution. You probably need to have 'make'. See /home/zm/.cpanm/work/1603109888.4775/build.log for details.
! Installing the dependencies failed: Module 'YAML::Tiny' is not installed
! Bailing out the installation for Module-Install-1.19.
```


I will check that `cpanminus gettext autoconf automake build-essential` are indeed the minimum required packages for a build environment in Ubuntu.

Reply:
I confirm, this looks like the minimum requirements.